new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
```

Comment:
I'm actually just using `def self.inherited` on `Chef::Resource`, which seems to do the job.  I'd like descendant tracking to go away, but we have to do this in chess moves, or we get RFCs like the last one :)